### ISSUE TITLE
[Bug]: SortAndBind with IObservable<IComparer> not working as expected

### ISSUE THREAD

**Issue by DevEngineReq** [reactivemarbles/DynamicData#932]
### Describe the bug 🐞

If we use SortAndBind with an IObservable<IComparer> we have to use ObserveOn(RxApp.MainThreadScheduler) for the compare observable, otherwise the application will freeze. For the deprecated .Sort() method it was not required. 

This behavour can cause wired problems you switch from .Sort to SortAndBind and just copy the compare observable.

![image](https://github.com/user-attachments/assets/d96eaf50-dccb-4cab-b695-4ce0f2d094f7)

### Reproduction repository

https://github.com/reactivemarbles/DynamicData

### Expected behavior

See Descriptions

**Comment by JakenVeina** [comment]
I can agree this looks unintuitive, but it's very much to be expected, and is in fact a behavior you can get with the separate `.Sort()` and `Bind()` operator.

Using the `.SortAndBind()` operator as in your example is the equivalent of doing the following:

```cs
this.cacheObservable = this.userCache.Connect()
    .ObserveOn(RxApp.MainThreadScheduler)
    .Sort(sorterObservable)
    .Bind(this.users)
    .Subscribe();
```

The issue is that there are 2 separate streams feeding input into the `.Sort()` operation, the `.Connect()` stream, and `sorterObservable`. When `.Sort()` and `.Bind()` are separate, it's trivial to insert the proper `.ObserveOn()` after the `.Sort()`, when the two separate streams have been merged into just one. But without that, you need to apply `.ObserveOn()` to both input streams, separately.

It sounds like we should add an optional `scheduler` parameter to `.SortAndBind()` to support this scenario, and to help clarify for folks migrating to it that a `scheduler` may be necessary.

**Comment by DevEngineReq** [comment]
Like this

![image](https://github.com/user-attachments/assets/390b51e7-80f9-4123-aae8-a9ebc3b34e54)

It would be nice if the stream would go ahead with the entered scheduler after SortAndBind(....)

**Comment by JakenVeina** [comment]
Yes, you'd end up with the downstream being driven by that scheduler.

**Comment by DevEngineReq** [comment]
Any news on this issue?

Will there be an optional scheduler parameter to .SortAndBind() like described above?

**Comment by JakenVeina** [comment]
In fact, this has just been added and released. You can now supply an `IScheduler` to `.SortAndBind()` via the `options` parameter.

**Comment by DevEngineReq** [comment]
Works as expected. Thank you
